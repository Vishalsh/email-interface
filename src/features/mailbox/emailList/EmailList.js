import React from 'react';
import PropTypes from 'prop-types';

import EmailListItem from 'features/mailbox/emailListItem/EmailListItem';

const EmailList = (props) => {
  const {
    emails
  } = props;

  return (
    <ul>
      {
        emails.map((email) => (
          <EmailListItem key={email.id} email={email}/>
        ))
      }
    </ul>
  );
};

EmailList.propTypes = {
  emails: PropTypes.array.isRequired
};

export default EmailList;
