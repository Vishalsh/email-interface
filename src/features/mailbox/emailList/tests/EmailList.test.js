import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import EmailList from '../EmailList';

describe('EmailList', () => {
  const props = {
    emails: [{
      id: 1,
      subject: 'subject 1',
      from: 'MS Dhoni'
    }, {
      id: 1,
      subject: 'subject 2',
      from: 'Virat Kohli'
    }]
  };
  const component = shallow(<EmailList {...props} />);

  it('should render the component', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
});