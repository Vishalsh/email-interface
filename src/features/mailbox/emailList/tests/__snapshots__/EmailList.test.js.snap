// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailList should render the component 1`] = `
<ul>
  <EmailListItem
    email={
      Object {
        "from": "MS Dhoni",
        "id": 1,
        "subject": "subject 1",
      }
    }
    key="1"
  />
  <EmailListItem
    email={
      Object {
        "from": "Virat Kohli",
        "id": 1,
        "subject": "subject 2",
      }
    }
    key="1"
  />
</ul>
`;
